Library/Homebrew/utils/spdx.rb

```diff
@@ -90,14 +91,14 @@ def license_expression_to_string(license_expression, bracket: false, hash_type:
     when String
       license_expression
     when Symbol
-      license_expression.to_s.tr("_", " ").gsub(/\b(?<!\w['’`()])[a-z]/, &:capitalize)
+      LICENSEREF_PREFIX + license_expression.to_s.tr("_", "-").gsub(/\b(?<!\w['’`()])[a-z]/, &:capitalize)
```

Comment:
Do we want to add Homebrew in the name of this? It will increase the length though.

From ScanCode[^1], it looks like camelcase `LicenseRef-PublicDomain` and Fedora-specific `LicenseRef-Fedora-Public-Domain` are seen out in the wild. Ours will be `LicenseRef-Public-Domain` in current logic.

[^1]: https://scancode-licensedb.aboutcode.org/public-domain

Reply:
Probably should prefix Homebrew in at least the case of `:cannot_represent` given that really probably shouldn't e

For public domain I'm OK with either.

Do update `string_to_license_expression` to support the reverse of whatever we do here.

Reply:
Doing the same for all or none is probably simpler in terms of mapping.

Only tricky part of longer name is if we switch to using official GitHub license `org.opencontainers.image.licenses` (plural) which has a 256 character limit:
https://github.com/Homebrew/brew/blob/abc0584215659c5a8716d0798ea2801388d01368/Library/Homebrew/github_packages.rb#L309

---

`string_to_license_expression` actually never worked for symbols. Code using JSON API currently cannot re-map `Cannot Represent` to `:cannot_represent`.

Reply:
Yeah that GitHub Packages thing is an unfortunate typo.

> `string_to_license_expression` actually never worked for symbols. Code using JSON API currently cannot re-map `Cannot Represent` to `:cannot_represent`.

Yeah this is a bug, though perhaps easier to solve now if all symbols have a common prefix.

Reply:
For now, I'll append `Homebrew-` to prefix and we can decide later on if we want to change it. Going from more specific to less is probably safer for external tooling.

---

> Yeah that GitHub Packages thing is an unfortunate typo.

It had a benefit since we didn't trigger any failures from invalid SPDX (e.g. `Cannot Represent`).

---

> Yeah this is a bug, though perhaps easier to solve now if all symbols have a common prefix.

Will also need a bit of a rewrite as parsing is naive and thus causes following:
```
> puts JSON.pretty_generate(SPDX.string_to_license_expression("A AND B OR C AND D OR E"))
{
  "all_of": [
    "A",
    "AND",
    {
      "any_of": [
        "B",
        "OR",
        "C"
      ]
    },
    "AND",
    {
      "any_of": [
        "D",
        "OR",
        "E"
      ]
    }
  ]
}
```